### PR TITLE
Restructure toctree documentation

### PR DESCRIPTION
**Pull request by timhoffm**
The main change is structuring the options using `.. rst:directive:option::`. I've adapted some of their descriptions where suitable.

### CODE REVIEW

**Comment by timhoffm** [comment]
@picnixz thanks for the response. No hurry. It doesn't make a big difference whether this goes in a few days earlier or later 😄.

File: doc/usage/restructuredtext/directives.rst [hunk 1]
```diff
@@ -96,138 +96,128 @@ The ``toctree`` directive is the central element.
...
+      .. versionadded:: 0.6
 
-      .. toctree::
-         :glob:
-         :reversed:
+      .. versionchanged:: 1.1
+         Added numeric argument to "numbered".
 
-         recipe/*
+   .. rst:directive:option:: titlesonly
```
Comment (electric-coder) [inline doc/usage/restructuredtext/directives.rst:188]: Isn't this incorrect syntax? Why write it in a single line?

```rst
.. rst:directive::
   :option:
```

or 

```rst
.. toctree::
   :titlesonly:
```

Comment (timhoffm) [reply doc/usage/restructuredtext/directives.rst:188]: I've taken this from https://github.com/sphinx-doc/sphinx/blob/96b511798da0ab819b87be363fe505186896fee2/doc/usage/restructuredtext/directives.rst?plain=1#L665 and similar.

it's a bit peculiar naming, but `directive:option` is a directive in the`rst` domain:
https://github.com/sphinx-doc/sphinx/blob/96b511798da0ab819b87be363fe505186896fee2/sphinx/domains/rst.py#L229

Comment (electric-coder) [reply doc/usage/restructuredtext/directives.rst:188]: Ok, makes sense now :)

**Review by AA-Turner — APPROVED** [review]
A nice improvement. Thanks Tim!

A

**Pull request merged**